csharp/src/Ice/UdpEndpoint.cs

```diff
@@ -4,61 +4,51 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Globalization;
 using System.Net;
 using System.Text;
 
 namespace ZeroC.Ice
 {
     /// <summary>The Endpoint class for the UDP transport.</summary>
-    public sealed class UdpEndpoint : IPEndpoint
+    internal sealed class UdpEndpoint : IPEndpoint
     {
         // We cannot move HashCompressFlag to IPEndpoint because of the way we marshal it in TcpEndpoint
         public override bool HasCompressionFlag { get; }
         public override bool IsDatagram => true;
 
-        /// <summary>The local network interface used to send multicast datagrams.</summary>
-        public string McastInterface { get; } = "";
-
-        /// <summary>The time-to-live of the multicast datagrams, in milliseconds.</summary>
-        public int McastTtl { get; } = -1;
+        public override string? this[string option] =>
+            option switch
+            {
+                "interface" => McastInterface,
+                "ttl" => _mcastTtl.ToString(CultureInfo.InvariantCulture),
+                _ => base[option],
+            };
 
-        public override int Timeout => -1;
         public override Transport Transport => Transport.UDP;
 
+        /// <summary>The local network interface used to send multicast datagrams.</summary>
+        internal string McastInterface { get; } = "";
+
         private readonly bool _connect;
         private int _hashCode = 0;
 
+        /// <summary>The time-to-live of the multicast datagrams, in hops.</summary>
+        private int _mcastTtl { get; } = -1;
```

Comment:
It would be good to add a comment to explain why this one is private and McastInterface isn't.

Reply:
I think it's accidental. IceDiscovery and IceLocatorDiscovery use UdpEndpoint and the internal McastInterface property directly, they probably should just use the public API. What do you think?

Reply:
I'm not sure what you mean. Do you mean McastInterface should be public or that the discovery plugins should somehow figure it out using the public API (which one, ToString?) and mcast interface should be private like mcast ttl? These two attributes used to be part of the public API before.

Reply:
IceDiscovery and IceLocatorDiscovery could use Transport to figure out if an endpoint is UDP, and they could use `endpoint["interface"]` to retrieve this interface (as a string). I.e. they would not use the internal UdpEndpoint class or its properties anymore.

After merging in your PR, I am making the 2 UdpEndpoint properties internal since they are now used by UdpConnector.

Reply:
Ok sounds good.